### PR TITLE
Add support for Llama 3 on Amazon SageMaker

### PR DESCRIPTION
**Pull request by dylanholmes**
Changes:
- Rename `AmazonSageMakerPromptDriver.model` to `AmazonSageMakerPromptDriver.endpoint`.
- Edit `AmazonSageMakerPromptDriver.model` to represent the optional `InferenceComponentName` passed to `SageMakerRuntime.Client.invoke_endpoint`. `InferenceComponentName` is required by multi-model endpoints, which is what SageMaker JumpStart deploys (doesn't even have an option to deploy single-model endpoints).
- Edit `BasePromptModelDriver.process_output` (and subclasses) to allow the `output` param to be a `dict`, as this is the output returned by Llama 3 Instruct when deployed via Amazon SageMaker JumpStart.
- Add Llama 3 support to `SageMakerLlamaPromptModelDriver`.
- Remote Llama 2 support from `SageMakerLlamaPromptModelDriver`.
- Edit `Llama` example to the `Amazon SageMaker` section of the prompt driver docs.

Manual Example Testing:
1. Deploy Meta-Llama-3-8B-Instruct on SageMaker via JumpStart. (AWS Console > SageMaker > Studio > JumpStart > Meta > Meta-Llama-3-8B-Instruct > Deploy)
   1. Accept EULA
   1. Endpoint name: `Enter endpoint name`
   1. Custom endpoint name: `my-endpoint`
   1. Instance type: `ml.g5.2xlarge`
   1. Initial instance count: `1`
   1. Maximum instance count: `1`
   1. Number of accelerators: `1`
   1. Min number of copies: `1`
   1. Min CPU memory (MB): `128`
   1. Click deploy, then wait for the deployment to succeed and the model to be in the `In Service` state.
1. After deployment you should be automatically redirected to the endpoint details page. From there:
    1. Note down the region of your endpoint. (In the studio URL and endpoint ARN)
    1. Note down the endoint name (should be `my-endpoint` if you followed above).
    1. Note down the model name (probably something like `meta-textgeneration-llama-3-8b-instruct-xx-yy`)
1. Configure your AWS credentials (and region). (I used `aws sso login`)
1. Request access to [meta-llama/Meta-Llama-3-8B-Instruct](https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct) (for tokenizer usage).
1. Configure hugging face credentials in local test environment.
1. Run example code

Observed the following output:
```
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
[05/23/24 12:47:35] INFO     PromptTask c865b7ca66d94bcd813be8bc854cc655
                             Input: Hello!
[05/23/24 12:47:38] INFO     PromptTask c865b7ca66d94bcd813be8bc854cc655
                             Output: Arrrr, shiver me timbers! 'Tis a grand day to be havin' a chat, matey! What be bringin' ye to these fair waters?
```

---

Note:
- I noticed that the `Amazon SageMaker` prompt driver doc examples are all marked with `title="PYTEST_IGNORE"`. I'm assuming that we are doing this because we don't want to pay to keep an endpoint continually running, which would make sense. Therefore, I also marked this new example with `title="PYTEST_IGNORE"`.

----
📚 Documentation preview 📚: https://griptape--800.org.readthedocs.build//800/

### CODE REVIEW

File: griptape/drivers/prompt_model/sagemaker_jumpstart_llama3_instruct_prompt_model_driver.py
Comment (collindutter) [inline griptape/drivers/prompt_model/sagemaker_jumpstart_llama3_instruct_prompt_model_driver.py:1]: Can we somehow roll this functionality into `SageMakerLlamaPromptModelDriver`? Or should we rename `SageMakerLlamaPromptModelDriver` to be more consistent with this? I don't love either option tbh..

**Review by collindutter — COMMENTED** [review]
Functionality looks great, only concern is regarding naming

File: griptape/drivers/prompt_model/sagemaker_jumpstart_llama3_instruct_prompt_model_driver.py
Comment (dylanholmes) [reply griptape/drivers/prompt_model/sagemaker_jumpstart_llama3_instruct_prompt_model_driver.py:1]: As discussed, decided to replace support for Llama 2 with Llama 3 in the existing class.

File: docs/griptape-framework/drivers/prompt-drivers.md [hunk 1]
```diff
@@ -416,7 +431,8 @@ from griptape.config import StructureConfig
 agent = Agent(
     config=StructureConfig(
         prompt_driver=AmazonSageMakerPromptDriver(
-            model=os.environ["SAGEMAKER_LLAMA_ENDPOINT_NAME"],
+            endpoint=os.environ["SAGEMAKER_LLAMA_3_INSTRUCT_ENDPOINT_NAME"],
+            model=os.environ["SAGEMAKER_LLAMA_3_INSTRUCT_INFERENCE_COMPONENT_NAME"],
```
Comment (collindutter) [inline docs/griptape-framework/drivers/prompt-drivers.md:435]: Needs to be updated in https://github.com/griptape-ai/griptape/blob/81ec3b637ae9c924a80ecdfdfb472f760ef99ee1/.github/workflows/docs-integration-tests.yml#L80

File: docs/griptape-framework/drivers/prompt-drivers.md [hunk 2]
```diff
@@ -446,7 +462,8 @@ from griptape.config import StructureConfig
 agent = Agent(
     config=StructureConfig(
         prompt_driver=AmazonSageMakerPromptDriver(
-            model=os.environ["SAGEMAKER_FALCON_ENDPOINT_NAME"],
+            endpoint=os.environ["SAGEMAKER_FALCON_ENDPOINT_NAME"],
+            model=os.environ["SAGEMAKER_FALCON_INFERENCE_COMPONENT_NAME"],
```
Comment (collindutter) [inline docs/griptape-framework/drivers/prompt-drivers.md:466]: Samsies.

File: griptape/drivers/prompt_model/sagemaker_llama_prompt_model_driver.py [hunk 3]
```diff
@@ -8,34 +8,40 @@
 
 @define
 class SageMakerLlamaPromptModelDriver(BasePromptModelDriver):
-    DEFAULT_MAX_TOKENS = 600
+    # Default context length for all Llama 3 models is 8K as per https://huggingface.co/blog/llama3
+    DEFAULT_MAX_TOKENS = 8000
```
Comment (collindutter) [inline griptape/drivers/prompt_model/sagemaker_llama_prompt_model_driver.py:12]: Can you please rename this to `DEFAULT_MAX_INPUT_TOKENS` for consistency with the Tokenizer classes?

**Review by collindutter — CHANGES_REQUESTED** [review]
Just a couple minor things, otherwise looks good!

**Review by collindutter — APPROVED** [review]

**Review by collindutter — APPROVED** [review]

**Pull request merged**